receptorctl/receptorctl/socket_interface.py

```diff
@@ -98,9 +98,12 @@ def connect_to_service(self, node, service, tlsclient):
         if not str.startswith(text, "Connecting"):
             raise RuntimeError(text)
 
-    def submit_work(self, node, worktype, payload, tlsclient, ttl, params):
-        if node is None:
-            node = "localhost"
+    def submit_work(self, *, worktype, payload, node=None, tlsclient=None, ttl=None, params=None):
+        node = node or "localhost"
+        tlsclient = tlsclient or ''
+        ttl = ttl or ''
+        params = params or ''
```

Comment:
I think it would be better to conditionally add these optional parameters to commandMap rather than setting them to the empty string.  If params is None then the `if params` below will skip the section, for example.  And the rest are optional and don't need to be in the json if we aren't using them.

Reply:
@ghjm Done. How about now?

Reply:
Yep, this looks good, thanks!